includes/wizards/class-setup-wizard.php

```diff
@@ -0,0 +1,144 @@
+<?php
+/**
+ * Newspack setup wizard. Required plugins, introduction, and data collection.
+ *
+ * @package Newspack
+ */
+
+namespace Newspack;
+
+use \WP_Error, WP_REST_Server;
+defined( 'ABSPATH' ) || exit;
+require_once NEWSPACK_ABSPATH . '/includes/wizards/class-wizard.php';
+
+define( 'NEWSPACK_SETUP_COMPLETE', 'newspack_setup_complete' );
+
+/**
+ * Setup Newspack.
+ */
+class Setup_Wizard extends Wizard {
+	/**
+	 * The slug of this wizard.
+	 *
+	 * @var string
+	 */
+	protected $slug = 'newspack-setup-wizard';
+	/**
+	 * The capability required to access this wizard.
+	 *
+	 * @var string
+	 */
+	protected $capability = 'edit_products';
+
+	/**
+	 * Display a link to this wizard in the Newspack submenu.
+	 *
+	 * @var bool
+	 */
+	protected $hidden = false;
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		parent::__construct();
+		add_action( 'current_screen', [ $this, 'redirect_to_setup' ] );
+		add_action( 'rest_api_init', [ $this, 'register_api_endpoints' ] );
+	}
+
+	/**
+	 * Get the name for this wizard.
+	 *
+	 * @return string The wizard name.
+	 */
+	public function get_name() {
+		return esc_html__( 'Setup', 'newspack' );
+	}
+	/**
+	 * Get the description of this wizard.
+	 *
+	 * @return string The wizard description.
+	 */
+	public function get_description() {
+		return esc_html__( 'Setup Newspack.', 'newspack' );
+	}
+	/**
+	 * Get the expected duration of this wizard.
+	 *
+	 * @return string The wizard length.
+	 */
+	public function get_length() {
+		return esc_html__( '10 minutes', 'newspack' );
+	}
+
+	/**
+	 * Register the endpoints needed for the wizard screens.
+	 */
+	public function register_api_endpoints() {
+		// Update option when setup is complete.
+		register_rest_route(
+			'newspack/v1/wizard/' . $this->slug,
+			'/complete',
+			[
+				'methods'             => WP_REST_Server::EDITABLE,
+				'callback'            => [ $this, 'api_complete' ],
+				'permission_callback' => [ $this, 'api_permissions_check' ],
+			]
+		);
+	}
+
+	/**
+	 * Update option when setup is complete.
+	 *
+	 * @return WP_REST_Response containing info.
+	 */
+	public function api_complete() {
+		update_option( NEWSPACK_SETUP_COMPLETE, true );
+		return rest_ensure_response( [ 'complete' => true ] );
+	}
+
+	/**
+	 * Enqueue Subscriptions Wizard scripts and styles.
+	 */
+	public function enqueue_scripts_and_styles() {
+		parent::enqueue_scripts_and_styles();
+		wp_enqueue_media();
```

Comment:
We don't need `wp_enqueue_media` unless there's an image uploader (it enqueues the media modal JS). Sorry, I probably should have documented that better.

Reply:
Good to know. https://github.com/Automattic/newspack-plugin/pull/114/commits/40ab8d905ccc751eb37e269dffbd00ccba19cfe6